src/Microsoft.Health.Fhir.Liquid.Converter/Parsers/JsonDataParser.cs

```diff
@@ -4,15 +4,35 @@
 // -------------------------------------------------------------------------------------------------
 
 using System;
+using System.IO;
+using EnsureThat;
 using Microsoft.Health.Fhir.Liquid.Converter.Exceptions;
 using Microsoft.Health.Fhir.Liquid.Converter.Extensions;
 using Microsoft.Health.Fhir.Liquid.Converter.Models;
+using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Health.Fhir.Liquid.Converter.Parsers
 {
     public class JsonDataParser : IDataParser
     {
+        private static Func<string, JsonReader> _defaultJsonReaderGenerator = (json) => new JsonTextReader(new StringReader(json))
+        {
+            DateParseHandling = DateParseHandling.None,
```

Comment:
@dustinburson  @pallar-ms This is technically a breaking change for anyone using the OSS code. But I'd like to address it here as its a bug. A couple of ideas:

- Bump the major version of the oss code to account for the breaking change
- Keep the existing behavior in OSS, and create different implementations in PaaS products which set this value to `DateParseHandling.None`.

Thoughts?

Reply:
Have you looked at how this is defined/used in the FHIR service.  I am concerned this won't allow us to change the behavior per request at the FHIR service level.

I am not concerned about a breaking change in the OSS.  We can do the necessary version update as you said and add documentation if necessary.

Reply:
Is the plan to use an API version of the request or another request body input to decide which parser to use?
Btw the convert G2 also loads the processors into the map during app init and on request, depending on the input data type selects the processor. https://microsofthealth.visualstudio.com/Health/_git/convert?path=/convert/core/src/Microsoft.Health.Convert.LiquidConverter/Handlers/LiquidConverterHandler.cs
We can update the key tuple with another value(whatever the new input we are basing it off of) and update the map that way too.

Reply:
Sure, that would work and I agree keeping it in the factory would be better. Might need to adjust this accordingly - https://github.com/microsoft/FHIR-Converter/blob/main/src/Microsoft.Health.Fhir.Liquid.Converter/Processors/ConvertProcessorFactory.cs

The only reason I proposed the map update was to keep the pattern of choosing the processor consistent from the map. Otherwise on first glance it looks odd why for one request param 'JsonFhirConversionDatesAsStrings' we pick the processor differently and then for another request param 'InputDataType' we use the map. But not strongly advocating it either since yeah adding to the key tuple is not too extendable  and neat if we have another field later.